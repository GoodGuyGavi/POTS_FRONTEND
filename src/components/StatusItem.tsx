import React from "react";
import { Label, Table, Divider } from "semantic-ui-react";
import StatusList from "../components/StatusList";

const yellow = ["yellow", "pink", "red"];
const pink = ["pink"];
const red = ["red"];

export interface IStatusItemProps {
  OR?: any;
  label?: any;
  status?: any;
  label3?: any;
  number?: any;
  statOption?: any;
  statuslist?: any;
  title?: any;
}

const StatusItem = (props: IStatusItemProps) => {
  const {
    OR,
    label,
    status,
    label3,
    number,
    statOption,
    statuslist,
    title
  } = props;
  return (
    <div>
      <StatusList />
      {props.statuslist.map((list: any) => (
        <Table.Row>
          <Label size="big" style={{ marginLeft: 20 }} color="green">
            {list.OR}
          </Label>

          <text style={{ marginLeft: 10 }}>{list.label}</text>

          <Divider section />
        </Table.Row>
      ))}
    </div>
  );
};

export default StatusItem;
