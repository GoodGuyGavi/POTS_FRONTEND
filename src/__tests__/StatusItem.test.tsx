/**
 * @format
 */

import React from "react";
import { shallow } from "enzyme";
import StatusItem from "../components/StatusItem";

const props = {
  statuslist: [
    {
      OR: "3",
      label: "Orders Received"
    },
    {
      OR: "7",
      label: "Released from supplier "
    },
    {
      OR: "4",
      label: "On-hold from supplier"
    }
  ]
};

describe(" Status Item", () => {
  test("renders correctly", () => {
    expect(shallow(<StatusItem {...props} />)).toMatchSnapshot();
  });
});
