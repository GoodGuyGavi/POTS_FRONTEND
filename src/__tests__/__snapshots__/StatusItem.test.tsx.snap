// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Status Item renders correctly 1`] = `ShallowWrapper {}`;
